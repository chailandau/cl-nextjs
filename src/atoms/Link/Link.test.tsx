import { composeStory } from '@storybook/react';
import { render, screen } from '@testing-library/react';

import Meta, { Default } from './Link.stories';

import {
    testAxeAndSnapshot,
    testKeyDown,
    testPropOptions,
    testRenderText
} from '@/utils/testHelpers';

const Link = composeStory(Default, Meta);
const linkComponent = {
    component: <Link />
};

describe('Link', () => {
    it('defaults correctly', () => {
        render(<Link underline={undefined} href={''} />);
        expect(screen.getByRole('link')).toHaveClass('underline');
        expect(screen.getByRole('link')).toHaveAttribute('href', '');
    });
    testPropOptions({
        ...linkComponent,
        propName: 'className',
        propOptions: ['test'],
        htmlTag: 'a'
    });
    testRenderText({
        ...linkComponent,
        text: (Link.args?.children as string) + '>',
        role: 'link'
    });
    testKeyDown({ ...linkComponent, role: 'link' });
    testAxeAndSnapshot(linkComponent);
});
