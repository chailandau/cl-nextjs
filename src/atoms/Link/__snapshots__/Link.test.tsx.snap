// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex"
  >
    <a
      class="link"
      href="/"
      rel=""
      target=""
    >
      <span
        class="link-text"
      >
        This is a link
      </span>
      <span
        class="link-icon"
      >
        &gt;
      </span>
    </a>
  </div>
</DocumentFragment>
`;
